.github/workflows/deploy-stage.yml

```diff
@@ -0,0 +1,24 @@
+---
+name: Deploy Stage
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    name: Deploy
+    steps:
+      - uses: actions/checkout@v2
+      - name: Publish
+        uses: cloudflare/wrangler-action@2.0.0
+        with:
+          apiToken: ${{ secrets.CF_API_TOKEN }}
+          accountId: ${{ secrets.CF_ACCOUNT_ID }}
+          secrets: |
+            OPEN_WEATHER_API_KEY
+          command: publish --env stage
+        env:
+          OPEN_WEATHER_API_KEY: ${{ secrets.OPEN_WEATHER_API_KEY }}
```

Comment:
Please change `secrets.OPEN_WEATHER_API_KEY` to `secrets.OPEN_WEATHER_API_KEY_STAGE`.